src/@providers/docker/services/deployment.ts

```diff
@@ -211,129 +211,7 @@ export class DockerDeploymentService extends CrudResourceService<'deployment', D
-      );
-    }
-
-    const exposedPorts = inputs.exposed_ports || [];
-    if (!inputs.exposed_ports) {
-      Object.keys(inspection.HostConfig.PortBindings).forEach((existingPort) => {
-        const [targetPort] = existingPort.split('/');
-        const hostPort = inspection.HostConfig.PortBindings[existingPort][0].HostPort;
-        exposedPorts.push({
-          port: Number(hostPort),
-          target_port: Number(targetPort),
-        });
-      });
-    }
-
-    for (const port of exposedPorts) {
-      let value = String(port!.target_port);
-      if (port?.port) {
-        value = `${port.port}:${value}`;
-      }
-      args.push('-p', value);
-    }
-
-    let labels = inspection.Config.Labels;
-    if (inputs.labels && inputs.name) {
-      labels = {
-        ...inputs.labels,
-        'io.architect': 'arcctl',
-        'io.architect.arcctl.deployment': containerName,
-      };
-    } else if (inputs.labels) {
-      labels = {
-        ...labels,
-        ...inputs.labels as Record<string, string>,
-      };
-    }
-
-    for (const [key, value] of Object.entries(labels)) {
-      args.push('--label', `${key}=${value}`);
-    }
-
-    if (inputs.volume_mounts) {
-      for (const mount of inputs.volume_mounts) {
-        args.push('--volume', `${mount?.volume}:${mount?.mount_path}`);
-      }
-    }
-
-    args.push(inputs.image || inspection.Image);
-
-    const command = (inputs.command || inspection.Config.Cmd) as string | string[];
-    if (command) {
-      args.push(...(typeof command === 'string' ? [command] : command));
-    }
-
-    const { code, stderr } = await exec('docker', { args });
-    if (code !== 0) {
-      throw new Error(stderr || 'Deployment failed');
-    }
-
-    const inspectionRes = await this.inspect(containerName);
-    const networks = Object.keys(inspectionRes?.NetworkSettings.Networks || {});
-    const ipAddress = inspectionRes?.NetworkSettings.Networks[networks[0]].IPAddress;
-
-    for (const serviceConfig of (inputs.services as ResourceInputs['deployment']['services'] || [])) {
-      const writableService = await this.providerStore.getWritableService(serviceConfig.account, 'service');
-      const existingService = await writableService.get(serviceConfig.id);
-      if (existingService?.target_servers) {
-        const newUrl = `http://${ipAddress}:${serviceConfig.port}`;
-
-        const target_servers = existingService.target_servers;
-        if (!target_servers.includes(newUrl)) {
-          target_servers.push(newUrl);
-        }
-
-        await lastValueFrom(writableService.apply({
-          ...existingService,
-          type: 'service',
-          target_servers,
-        }, {
-          id: serviceConfig.id,
-          providerStore: this.providerStore,
-        }));
-      }
-    }
-
-    return {
-      id: containerName,
-      labels,
-    };
+    return this.create(subscriber, inputs as ResourceInputs['deployment']);
```

Comment:
@mueschm the reason I didn't do this was for the case where we'd partially update an existing instance, but we don't even have an `update` CLI command right now. I'll approve this, but this would prevent discrete field updates.